src/Qowaiv.DomainModel/Collections/ImmutableCollection.Append.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace Qowaiv.DomainModel.Collections;
+
+public partial class ImmutableCollection
+{
+    /// <remarks>Internal from .NET.</remarks>
+    private const int MaxCapacity = 0X7FFFFFC7;
+
+    [Pure]
+    private ImmutableCollection Append(IEnumerator iterator)
+    {
+        var count = Count;
+        var capacity = Capacity;
+        var buffer = Buffer;
+        var locker = Locker;
+        var clone = true;
+
+        if (count < capacity && buffer[count] is null)
+        {
+            lock (locker)
+            {
+                if (buffer[count] is null)
+                {
+                    while (count < capacity && iterator.MoveNext())
+                    {
+                        var item = iterator.Current;
+                        if (item is null) continue;
+                        buffer[count++] = item;
```

Comment:
It _might_ be ever so slightly faster if you just use `count` here and add a `++count` statement after, depending on if the compiler is capable of optimising this properly.